examples/image_processing/gradient_diffusion.cpp

```diff
@@ -0,0 +1,99 @@
+/*******************************************************
+ * Copyright (c) 2017, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <stdio.h>
+#include <arrayfire.h>
+#include <af/util.h>
+#include <cstdlib>
+
+using namespace af;
+
+static const unsigned ITERS = 64;
+
+array normalize(const array &p_in)
+{
+    float mx = max<float>(p_in);
+    float mn = min<float>(p_in);
+    return (p_in-mn)/(mx-mn);
+}
+
+array sobelFilter(const array &p_in)
+{
+    int w = 5;
+    if (p_in.dims(0) <  512) w = 3;
+    if (p_in.dims(0) > 2048) w = 7;
+
+    int h = 5;
+    if (p_in.dims(0) <  512) h = 3;
+    if (p_in.dims(0) > 2048) h = 7;
+
+    array ker = gaussianKernel(w, h);
+    array smooth = convolve(p_in, ker);
+
+    for (unsigned i=1; i<ITERS; ++i)
+        smooth = convolve(smooth, ker);
+
+    array Gx, Gy;
+    sobel(Gx, Gy, smooth, 3);
+
+    return normalize(hypot(Gx, Gy));
+}
+
+array in, edges, smoothed;
+
+void anisotropicSmoothing()
+{
+    smoothed = gradientAnisotropicDiffusion(in, 0.125, 0.35f, ITERS, AF_FLUX_EXPONENTIAL);
+}
+
+int main(int argc, char* argv[])
+{
+    int device = argc > 1 ? atoi(argv[1]) : 0;
+
+    try {
+        af::setDevice(device);
+        af::info();
```

Comment:
you don't need the af namespace because you are including the af namespace at the top.